docs/design/referral-manager.md

```diff
@@ -0,0 +1,206 @@
+
+# Title: Referral Manager
+
+## Abstract
+
+Right now, referral program only works per satellite meaning a user on us-east-1 satellite can only refer people to join us-ease-1. 
+However, it's crucial for the tardigrade branded satellites to be able to handle cross-satellite referrals. Therefore, we plan to have a program running in GCP along with tardigrade satellites to handle this issue.
+
+## Background
+
+Along with cross-satellite referrals, we are also working on having a sign-up page that will allow users to select which satellite
+they would like to register an account with. Below design is built based on the new sign-up flow.
+
+## Design
+
+####Definition
+*Referral Link* - _A static link that contains a unique referral code_
+
+*Referral Code* - _An uuid generated by referral manager program for each user to be able to participate in our referral program_
+
+#### User Flow
+1. Sign up
+
+    After new user Alice chose to register on satellite B and clicked on the create account button:
+    - Satellite B sends a request with its satellite id to referral manager for a referral code for Alice
+    - Referral manager receives the request from satellite B, generates an unique referral code 11111111, and then store that code, 11111111, along with the satellite B's address into referral manager's database 
+    - After successfully storing the referral code and satellite B's address, referral manager sends a response with the generated referral code, 11111111, back to satellite B 
+    - Satellite B receives the referral code, 11111111, from referral manager and then store it along with Alice's information into user table in satellitedb
+        - Signed up WITHOUT referral link:
+            - Satellite B will store Alice's credit into user_credit table with a null value in referred_by column
+        - Signed up WITH referral link:
+            - If user Bonnie from satellite C referred Alice with a referral link, https://C.tardigrade.io/signup/22222222
```

Comment:
Just ensuring, the token can be used on whatever satellite i want? How would Bonnie know the Satellite that Alice wants to use? In this example she would refer to the same satellite as she is on. Wouldnt it need to be a "general" valid sign up page, something like https://signup.tardigrade.io/signup/222222 and there Alice can chose her satellite?

Reply:
Yes, we are developing a generic sign up page that has satellite selection in it